test/e2e/framework/job/wait.go

```diff
@@ -82,14 +82,30 @@ func waitForJobPodsInPhase(ctx context.Context, c clientset.Interface, ns, jobNa
 // both conformance CI jobs with GA-only features and e2e CI jobs with all default-enabled features.
 // So, we need to skip "Complete" condition reason verifications in the e2e conformance test cases.
 func WaitForJobComplete(ctx context.Context, c clientset.Interface, ns, jobName string, reason *string, completions int32) error {
-	if err := wait.PollUntilContextTimeout(ctx, framework.Poll, JobTimeout, false, func(ctx context.Context) (bool, error) {
-		curr, err := c.BatchV1().Jobs(ns).Get(ctx, jobName, metav1.GetOptions{})
+	err := framework.Gomega().Eventually(ctx, framework.HandleRetry(func(ctx context.Context) (*batchv1.Job, error) {
```

Comment:
I'm not very familiar with `HandleRetry` and `MakeMatcher`, I suppose they are used to fail fast?

I'm wondering if there is a less-code alternative: await for the function used in eventually to return the number of succeeded pods, and use gomega to check `Should(gomega.Equal(completions))`. Now, if the opposite condition happens, then use framework.Fail to fail fast?

Reply:
I did research alternative implementation and came up with a new proposal based on your suggestions which is indeed less verbose. PTAL and let me know your thoughts 😊